LayoutTests/ipc/invalid-svgfilter-expression-crash.html

```
@@ -0,0 +1,95 @@
+
+    const remoteImageBuffer = streamConnection.newInterface("RemoteImageBuffer", imageBufferIdentifier);
+
+    try {
+        remoteImageBuffer.FilteredNativeImage({
+            filter: {
+                subclasses: {
+                    variantType: 'WebCore::SVGFilter',
+                    variant: {
+                        targetBoundingBox: {
+                            location: {x: 0, y: 0},
+                            size: {width: 0, height: 0},
+                        },
+                        primitiveUnits: 0,
+                        expression: {
+                            alias: [
+                                {index: 1337, level: 0, geometry: 0}    // Adds an expression with an OOB index
+                            ]
+                        },
+                        effects: [
+                            {
+                                subclasses: {
+                                    variantType: 'WebCore::FEBlend',
+                                    variant: {
+                                        blendMode: 8,
+                                        operatingColorSpace: {serializableColorSpace: {alias: {m_cgColorSpace: {alias: {variantType: 'WebCore::ColorSpace', variant: 1}}}}}
+                                    }
+                                }
+                            }
+                        ],
+                        renderingResourceIdentifierIfExists: {},
+                        filterRenderingModes: 1,
+                        filterScale: {width: 0, height: 0},
+                        filterRegion: {
+                            location: {x: 0, y: 0},
+                            size: {width: 0, height: 0}
+                        },
+                    }
+                },
+            }
+        });
+    } catch(err) {
+        if(!(err instanceof TypeError)) {
+            console.log("Expected TypeError, got " + err);
```

Is it expected that WebKit does not crash and no exception was thrown? So how can we confirm we validated the expression in this test?

---

With the validation in place, it's expected that we get a TypeError indicating deserialisation failed. I wanted to log (and fail the test) if we get any other kind of error, as that might indicate the message wasn't sent and so the test could be non-functional. It's not perfect, since I think a change in message structure would also return the same TypeError. I'd be happy to hear any recommendations you might have to better detect the difference between and invalid message structure and a validation failure.

---

As we introduce more validator tests I suspect this might become more of an issue, so I filed an enhancement radar (rdar://147337600) for the IPC testing API to be able to report validation errors separately to deserialisation errors.

---

In the short-term, I added some additional comments to explain why I am checking for TypeError here, and made the log message more clear that anything else should be considered a test failure rather than errand logging left in from debugging. Once the enhancement radar is fixed, we should update the test to make sure the error came specifically from a validation error.